aws/lambda/pytorch-auto-revert/pytorch_auto_revert/workflow_resolver.py

```diff
@@ -102,3 +118,87 @@ def _build_indices(self) -> None:
                     ref = WorkflowRef(display_name=name, file_name=base)
                     self._by_display[name] = ref
                     self._by_file[base] = ref
+
+    def get_input_support(self, workflow_name: str) -> WorkflowInputSupport:
```

Comment:
remove the overcomplicated `self._input_support_cache` and replace with 

```
@lru_cache(maxsize=512)
def get_input_support(self, workflow_name: str) -> WorkflowInputSupport:
   # move the logic from self._fetch_and_parse_workflow_inputs here
```

Reply:
```

>>> Lint for aws/lambda/pytorch-auto-revert/pytorch_auto_revert/workflow_resolver.py:

  Warning (FLAKE8) B019
    Use of `functools.lru_cache` or `functools.cache` on methods can lead to
    memory leaks. The cache may retain instance references, preventing garbage
    collection.
    See

        130  |        ref = self.require(workflow_name)
        131  |        return self._fetch_workflow_input_support(ref.file_name)
        132  |
    >>> 133  |    @lru_cache(maxsize=None)
        134  |    def _fetch_workflow_input_support(self, file_name: str) -> WorkflowInputSupport:
        135  |        """Fetch and parse workflow input support with caching.
        136  |

```